operator/controllerutil/connector_common.go

```diff
@@ -0,0 +1,62 @@
+package controllerutil
+
+import (
+	"context"
+	"fmt"
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	providerv1 "github.com/vshn/provider-exoscale/apis/provider/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// ExoscaleAPIKey identifies the key in which the API Key of the exoscale.com is expected in a Secret.
+	ExoscaleAPIKey = "EXOSCALE_API_KEY"
+	// ExoscaleAPISecret identifies the secret in which the API Secret of the exoscale.com is expected in a Secret.
+	ExoscaleAPISecret = "EXOSCALE_API_SECRET"
+)
+
+type GenericConnector struct {
+	Kube     client.Client
+	Recorder event.Recorder
+}
+
+type GenericConnectContext struct {
+	context.Context
+	ApiKey             string
+	ApiSecret          string
+	ProviderConfigName string
+
+	providerConfig   *providerv1.ProviderConfig
+	credentialSecret *corev1.Secret
+}
+
+func (c *GenericConnector) FetchProviderConfig(ctx *GenericConnectContext) error {
+	if ctx.ProviderConfigName == "" {
+		return fmt.Errorf(".spec.providerConfigRef.Name is required")
+	}
```

Comment:
Why don't we enforce this in the admission webhook for Create/Update operations?

Reply:
Updated the logic and moved this to the validation webhook